node/src/Pos/Block/Logic/Header.hs

```diff
@@ -112,23 +122,25 @@ classifyNewHeader (Right header) = do
                     -- It's questionable though, maybe we will change
                     -- this decision.
                     , vhpCurrentSlot = Nothing
-                    -- [CSL-1152] TODO:
-                    -- we don't do these checks, but perhaps we can.
-                    , vhpLeaders = Nothing
-                    , vhpMaxSize = Nothing
+                    , vhpLeaders = Just leaders
+                    , vhpMaxSize = Just maxBlockHeaderSize
                     , vhpVerifyNoUnknown = False
                     }
-                verRes = verifyHeader vhp (Right header)
-            in case verRes of
-                   VerSuccess        -> CHContinues
-                   VerFailure errors -> mkCHRinvalid errors
+            case verifyHeader vhp (Right header) of
+                VerFailure errors -> throwError $ mkCHRinvalid errors
+                _                 -> pass
+
+            dlgHeaderValid <- runDBCede $ runExceptT $ dlgVerifyHeader header
+            whenLeft dlgHeaderValid $ throwError . CHInvalid
+
+            pure CHContinues
         -- If header's parent is not our tip, we check whether it's
         -- more difficult than our main chain.
-        | tipHeader ^. difficultyL < header ^. difficultyL -> CHAlternative
+        | tipHeader ^. difficultyL < header ^. difficultyL -> pure CHAlternative
```

Comment:
So???
Once again, suppose someone who is not a leader sends us a more difficult block which isn't a continuation of our chain. It will trigger recovery which is bad.

Reply:
No, it will not trigger the recovery. It will add a certain task to retrievalworker with this `h`. We will only enter the recovery after we get first k+ headers that lead us from our tip to `h`.

Reply:
Well, I still think this check should be done, but I don't mind if it's done in a separate issue.

Reply:
I see no need for this change. I've already explained why the "problem" you're trying to fix is not crucial.